engines/bastion_katello/app/assets/javascripts/bastion_katello/capsule-content/capsule-content.controller.js

```diff
@@ -0,0 +1,167 @@
+ *   Provides the functionality for the capsule-content page.
+ */
+angular.module('Bastion.capsule-content').controller('CapsuleContentController',
+    ['$scope', '$urlMatcherFactory', '$location', 'translate', 'CapsuleContent', 'AggregateTask',
+    function ($scope, $urlMatcherFactory, $location, translate, CapsuleContent, AggregateTask) {
+
+        var urlMatcher = $urlMatcherFactory.compile("/smart_proxies/:capsuleId");
+        var capsuleId = urlMatcher.exec($location.path()).capsuleId;
+
+        var syncState = {
+            DEFAULT: 'DEFAULT',
+            SYNCING: 'SYNCING',
+            SYNC_TRIGGERED: 'SYNC_TRIGGERED',
+            CANCEL_TRIGGERED: 'CANCEL_TRIGGERED',
+            FAILURE: 'FAILURE',
+            set: function (state) {
+                return this.state = state;
+            },
+            is: function () {
+                return _.contains(arguments, this.state || this.DEFAULT);
+            }
+        }
+
+        function refreshSyncStatus() {
+            CapsuleContent.sync_status({id: capsuleId}, function (syncStatus) {
+                $scope.syncStatus = syncStatus;
+                if (syncStatus.active_sync_tasks.length > 0) {
+                    $scope.syncTask = aggregateTasks(syncStatus.active_sync_tasks);
+
+                } else if (syncStatus.last_failed_sync_tasks.length > 0) {
+                    $scope.syncTask = pickLastTask(syncStatus.last_failed_sync_tasks);
+                    $scope.syncErrorMessages = $scope.syncTask.humanized.errors;
+                }
+                $scope.syncState.set(stateFromTask($scope.syncTask));
+            });
+        }
+
+        function taskUpdated(task) {
+            if ($scope.syncTask.result !== undefined && $scope.syncTask.result !== 'pending'){
+                refreshSyncStatus();
+            }
+        }
+
+        function aggregateTasks(tasks) {
+            var taskIds = _.map(tasks, function (task) {
+                                return task.id;
+                            });
+            return AggregateTask.new(taskIds, taskUpdated);
+        }
+
+        function pickLastTask(tasks) {
+            var task = tasks.slice(-1)[0];
+            task.progressbar = {
+                value: task.progress * 100,
+                type: $scope.progressbarType(task)
+            }
+            return task;
+        }
+
+        function isTaskInProgress(task) {
+            return (task && (task.state === 'pending' || task.state === 'running'))
+        };
+
+        function stateFromTask(syncTask) {
+            if (isTaskInProgress(syncTask))
+                return syncState.SYNCING;
+            if (syncTask && syncTask.result !== 'success')
+                return syncState.FAILURE;
+            else
+                return syncState.DEFAULT;
+        };
+
+
+        $scope.syncState = syncState;
+
+        $scope.expandEnvironments = {};
+
+        refreshSyncStatus();
+
+        $scope.$on('$destroy', function () {
```

Comment:
Is there a function of ours generating this? or is this being generated outside of the code in this controller?